src/DynamoCoreWpf/Controls/ShortcutToolbar.xaml.cs

```diff
@@ -107,6 +116,108 @@ private void LoginButton_OnClick(object sender, RoutedEventArgs e)
                 }
             }
         }
+
+        public List<Control> AllChildren(DependencyObject parent)
+        {
+            var _list = new List<Control> { };
+            for (int i = 0; i < VisualTreeHelper.GetChildrenCount(parent); i++)
+            {
+                var _child = VisualTreeHelper.GetChild(parent, i);
+                if (_child is Control)
+                {
+                    _list.Add(_child as Control);
+                    _list.AddRange(AllChildren(_child));
+                }
+            }
+            return _list;
+        }
+
+        Button GetButton(string buttonKey)
+        {
+            if (this.shortcutBarItems.Count > 1)
+            {
+                try
+                {
+                    int buttonIndex = ShortcutBarItems.ToList().FindIndex(item => item.ShortcutToolTip.ToUpper().Contains(buttonKey));
+                    var _container = ShortcutItemsControl.ItemContainerGenerator.ContainerFromIndex(buttonIndex);
+                    var _children = AllChildren(_container);
+                    var _control = (Button)_children.First();
+                    return _control;
+                }
+                catch (System.Exception)
+                {
+                    return null;
+                }                
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        public bool IsNewButtonEnabled
```

Comment:
Please use internal if they do not have to be public

Reply:
Done.